extensions/csrf-reactive/runtime/src/main/java/io/quarkus/csrf/reactive/runtime/CsrfRequestResponseReactiveFilter.java

```diff
@@ -230,7 +230,9 @@ private void createCookie(String csrfToken, RoutingContext routing, CsrfReactive
         if (config.cookieDomain.isPresent()) {
             cookie.setDomain(config.cookieDomain.get());
         }
-        routing.response().addCookie(cookie);
+        if (!routing.response().headWritten()) {
+            routing.response().addCookie(cookie);
```

Comment:
I'm not sure but shouldn't this cookie always be written in all requests?

@FroMage you confirm that when you have these errors, you don't care about the CSRF cookie?